HIRS_Utils/src/main/java/hirs/tpm/eventlog/TpmPcrEvent.java

```diff
@@ -0,0 +1,178 @@
+package hirs.tpm.eventlog;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+
+import hirs.utils.HexUtils;
+
+/**
+ * Class to process a TCG_PCR_EVENT.
+ * TCG_PCR_EVENT is used when the Event log uses the SHA1 Format as described in the
+ * TCG Platform Firmware Profile (PFP) specification.
+ *  typedef struct {
+ *    TCG_PCRINDEX  PCRIndex;  //PCR Index value that either
+ *                             //matches the PCRIndex of a
+ *                             //previous extend operation or
+ *                             //indicates that this Event Log
+ *                             //entry is not associated with
+ *                             //an extend operation
+ *    TCG_EVENTTYPE EventType; //See Log event types defined in toStrng()
+ *    TCG_DIGEST    digest;    //The hash of the event data
+ *    UINT32        EventSize; //Size of the event data
+ *    UINT8         Event[EventSize];  //The event data
+ * } TCG_PCR_EVENT;
+ */
+public class TpmPcrEvent {
+  /** Type length = 4 bytes. */
+  public static final int EV_TYPE_SIZE = 4;
+  /** Event Log spec version. */
+  public static final int MIN_SIZE = 32;
+  /** Event Type (byte array). */
+  public static final int INT_LENGTH = 4;
+  /** Event Type (byte array). */
+  public static final int SHA1_LENGTH = 20;
+  /** Event Type (byte array). */
+  public static final int SHA256_LENGTH = 32;
+  /** Each PCR bank holds 24 registers. */
+  public static final int PCR_COUNT = 24;
+  /** PCR index. */
+  private int pcrIndex = -1;
+  /** Event Type (long). */
+  private long eventType = 0;
+  /** Event digest. */
+  private byte[] digest = null;
+  /** Even data. */
+  private byte[] eventContent;
+  /** TCG Event Log spec version. */
+  private static String version = "Unknown";
+  /** TCG Event Log errata version. */
+  private static String errata = "Unknown";
+  /** Length (in bytes) of a pcr. */
+  private int digestLength = 0;
+
+  /**
+  * Constructor.
+  * @param is ByteArrayInputStream holding the event
+  * @throws IOException when event can't be parsed
+  */
+  public TpmPcrEvent(final ByteArrayInputStream is) throws IOException {
+
```

Comment:
I was looking at this and wondering why you needed an empty constructor that the children call, but the parameter isn't stored or used.

Reply:
Right now the Event log is only being processed for expected PCR values. In a future PR we will be adding capability to process individual events. This is not technically needed right now but is placed here in anticipation of that update.